binding_c needs "proper" prototypes

## Required information

**Operating system:**
E.g. maxOS 12.2.1

**Compiler version:**
E.g. Apple clang version 13.0.0 (clang-1300.0.29.30)

**Observed result or behaviour:**
Compiler warnings when compiling Cyclone DDS with Iceoryx 2.0 branch:
```
/Users/erik/C/iceoryx/install/include/iceoryx/v1.91.0/iceoryx_binding_c/config.h:23:32: warning: this function declaration is not a prototype [-Wstrict-prototypes]
uint32_t iox_cfg_max_publishers();
                               ^
                                void
```

**Expected result or behaviour:**
No warnings

**Conditions where it occurred / Performed steps:**
Build Cyclone DDS including https://github.com/eclipse-cyclonedds/cyclonedds/pull/1175 with default compilation setting on aforementioned platform.

--- a/iceoryx_binding_c/include/iceoryx_binding_c/config.h
+++ b/iceoryx_binding_c/include/iceoryx_binding_c/config.h
@@ -20,79 +20,79 @@
 #include <stdint.h>
 
 /// @brief returns maximum supported amount of publishers
-uint32_t iox_cfg_max_publishers();
+uint32_t iox_cfg_max_publishers(void);
 
 /// @brief returns maximum amount of subscribers which can be subscribed to one publisher
-uint32_t iox_cfg_max_subscribers_per_publisher();
+uint32_t iox_cfg_max_subscribers_per_publisher(void);
 
 /// @brief returns maximum amount of samples a publisher can acquire at the same time with loan
-uint32_t iox_cfg_max_chunks_allocated_per_publisher_simultaneously();
+uint32_t iox_cfg_max_chunks_allocated_per_publisher_simultaneously(void);
 
 /// @brief returns maximum history size for a publisher (e.g. samples which are hold back so
 //         that new subscribers can acquire past data)
-uint64_t iox_cfg_max_publisher_history();
+uint64_t iox_cfg_max_publisher_history(void);
 
 /// @brief returns maximum supported amount of subscribers
-uint32_t iox_cfg_max_subscribers();
+uint32_t iox_cfg_max_subscribers(void);
 
 /// @brief returns the maximum amount of samples which a subscriber can hold without releasing them
-uint32_t iox_cfg_max_chunks_held_per_subscriber_simultaneously();
+uint32_t iox_cfg_max_chunks_held_per_subscriber_simultaneously(void);
 
 /// @brief returns the maximum subscriber queue capacity which is used when the publisher delivers samples to the
 /// subscriber if the queue capacity is reached new samples will discard old samples
-uint32_t iox_cfg_max_subscriber_queue_capacity();
+uint32_t iox_cfg_max_subscriber_queue_capacity(void);
 
 /// @brief returns the maximum supported amount of condition variables. this determines how many listeners and waitsets
 /// can be used in one iceoryx system
-uint32_t iox_cfg_max_number_of_condition_variables();
+uint32_t iox_cfg_max_number_of_condition_variables(void);
 
 /// @brief returns the maximum supported amount of notifiers per condition variable. this determines how many
 /// attachments a listener or waitset can have
-uint32_t iox_cfg_max_number_of_notifiers_per_condition_variable();
+uint32_t iox_cfg_max_number_of_notifiers_per_condition_variable(void);
 
 /// @brief returns the maximum amount of attachments per waitset
 /// @note is less or equal to iox_cfg_max_number_of_notifiers_per_condition_variable
-uint32_t iox_cfg_max_number_of_attachments_per_waitset();
+uint32_t iox_cfg_max_number_of_attachments_per_waitset(void);
 
 /// @brief returns the maximum amount of evens per listener
 /// @note is less or equal to iox_cfg_max_number_of_notifiers_per_condition_variable
-uint32_t iox_cfg_max_number_of_events_per_listener();
+uint32_t iox_cfg_max_number_of_events_per_listener(void);
 
 /// @brief returns the maximum amount of mempools for roudi. restricts also the number of mempools in the roudi config
 /// file
-uint32_t iox_cfg_max_number_of_mempools();
+uint32_t iox_cfg_max_number_of_mempools(void);
 
 /// @brief returns the maximum number of shared memory segments. restricts also the number of configurable segments in
 /// the roudi config file
-uint32_t iox_cfg_max_shm_segments();
+uint32_t iox_cfg_max_shm_segments(void);
 
 /// @brief returns the maximum supported amount of shared memory providers
-uint32_t iox_cfg_max_number_of_memory_provider();
+uint32_t iox_cfg_max_number_of_memory_provider(void);
 
 /// @brief returns the maximum supported amount of memory blocks per shared memory provider
-uint32_t iox_cfg_max_number_of_memory_blocks_per_memory_provider();
+uint32_t iox_cfg_max_number_of_memory_blocks_per_memory_provider(void);
 
 /// @brief returns the alignment of the user payload when it is not set explicitly
-uint32_t iox_cfg_chunk_default_user_payload_alignment();
+uint32_t iox_cfg_chunk_default_user_payload_alignment(void);
 
 /// @brief returns the size of the user header when no user header is requested by the user
-uint32_t iox_cfg_no_user_header_size();
+uint32_t iox_cfg_no_user_header_size(void);
 
 /// @brief returns the alignment of the user header when no user header is requested by the user
-uint32_t iox_cfg_no_user_header_alignment();
+uint32_t iox_cfg_no_user_header_alignment(void);
 
 /// @brief returns the maximum supported amount of processes which can register at roudi by initializing the posh
 /// runtime
-uint32_t iox_cfg_max_process_number();
+uint32_t iox_cfg_max_process_number(void);
 
 /// @brief returns the maximum number of services that are supported byt the service registry
-uint32_t iox_cfg_service_registry_capacity();
+uint32_t iox_cfg_service_registry_capacity(void);
 
 /// @brief returns the maximum number of services a findservice call can return
-uint32_t iox_cfg_max_findservice_result_size();
+uint32_t iox_cfg_max_findservice_result_size(void);
 
 /// @brief returns the maximum runtime name length
-uint32_t iox_cfg_max_runtime_name_length();
+uint32_t iox_cfg_max_runtime_name_length(void);
 
 /// @brief the maximum size of a node name string + \0 terminator
 #define IOX_CONFIG_NODE_NAME_SIZE 101

